src/extension/alterschema/linter/then_false.h

```diff
@@ -0,0 +1,29 @@
+class ThenFalse final : public SchemaTransformRule {
+public:
+  ThenFalse()
+      : SchemaTransformRule{
+            "then_false",
+            "`if: S, then: false` is logically equivalent to `not: S`"} {};
+
+  [[nodiscard]] auto
+  condition(const JSON &schema, const JSON &, const Vocabularies &vocabularies,
+            const SchemaFrame &, const SchemaFrame::Location &,
+            const SchemaWalker &, const SchemaResolver &) const
+      -> SchemaTransformRule::Result override {
+    return contains_any(
+               vocabularies,
+               {"https://json-schema.org/draft/2020-12/vocab/applicator",
+                "https://json-schema.org/draft/2019-09/vocab/applicator",
+                "http://json-schema.org/draft-07/schema#"}) &&
+           schema.is_object() && schema.defines("if") &&
```

Comment:
Same comments as in the other `else_false` PR. I think you will have to check `else` to prevent semantic changes? I think you need to test those too